leap_drops/ProduceConfig.py

```diff
@@ -0,0 +1,129 @@
+import csv
+import json
+import math
+
+from dlg.droputils import DROPFile
+from dlg.drop import BarrierAppDROP
+from dlg.meta import dlg_int_param, dlg_float_param, dlg_string_param, \
+    dlg_component, dlg_batch_input, dlg_batch_output, dlg_streaming_input
+from dlg.droputils import DROPFile
+
+
+## Produce Config
+# @brief Produce Config
+# @details A BarrierAppDrop that produces multiple config files suitable for the CallLeap BarrierAppDrop
+# @par EAGLE_START
+# @param gitrepo $(GIT_REPO)
+# @param version $(PROJECT_VERSION)
+# @param category PythonApp
+# @param[in] param/number_of_stations/1/Integer
+#     \~English The number of stations from the measurement set that should be processed\n
+#     \~Chinese 要读取的起始频率\n
+#     \~
+# @param[in] param/implementation/eigen/String
+#     \~English The implementation of the LEAP algorithm to use (eigen, casa, cuda)\n
+#     \~Chinese 要读取的起始频率\n
+#     \~
+# @param[in] param/auto_correlation/false/string
+#     \~English Enable auto correlation in the LEAP algorithm\n
+#     \~Chinese 要读取的起始频率\n
+#     \~
+# @param[in] param/appclass/leap_nodes.ProduceConfig.ProduceConfig/String
+#     \~English The path to the class that implements this app\n
+#     \~Chinese 要读取的起始频率\n
+#     \~
+# @param[in] port/Directions
+#     \~English A CSV file containing directions for calibration
+#     \~Chinese 要读取的起始频率\n
+#     \~
+# @param[out] port/Config
+#     \~English A JSON config containing the specification for running an instance of LeapAccelerateCLI
+#     \~Chinese 要读取的起始频率\n
+#     \~
+# @par EAGLE_END
+
+class ProduceConfig(BarrierAppDROP):
+    """A BarrierAppDrop that produces multiple config files suitable for the CallLeap BarrierAppDrop"""
+    compontent_meta = dlg_component('ProduceConfig', 'Produce Config.',
+                                    [dlg_batch_input('binary/*', [])],
+                                    [dlg_batch_output('binary/*', [])],
+                                    [dlg_streaming_input('binary/*')])
+
+    numStations = dlg_int_param('number of stations', 1)
+    implementation = dlg_string_param('eigen', '')
+    autoCorrelation = dlg_int_param('auto correlation', 1)
+
+    # should be read from DALiuGE
+    #NUMBER_OF_COPIES = 1
+    #NUM_STATIONS = 126
+    #DIRECTIONS_FILENAME = "directions.csv"
+    #MEASUREMENT_SET_FILENAME = "/Users/james/working/leap-accelerate/testdata/1197638568-32.ms"
+    #IMPLEMENTATION = 'eigen'
```

Comment:
I should mention that I've renamed this implementation to 'cpu' in the command line arguments to make it clearer that it is the preferred implementation. Let me know support for the 'eigen' string is still required

Reply:
No, I'll update the components to use 'cpu' instead.